### PR TITLE
[py2py3] Migration at level scr/python/A/B/C - slice 2

### LINKED ISSUE #10098: [py2py3] Migration at level scr/python/A/B/C - slice 2
#### Intro

Our purpose is to continue with the gradual migration. This is the second slice in which the migration has been divided into. We have split migration units at level `src/python/A/B/C`.

A detailed plan of the migration at this depth is available at the dmwm mkdocs [detailed plan page](https://cms-dmwm-test.docs.cern.ch/py2py3/plan_detailed/), with summary information on top.

#### Impact of the new feature

We start migrating the second bulk of migration units at src/python/A/B/C level, which will include:

```plaintext
                                 name  #files  gradual  changes (+ and -)
          WMCore_ACDC_CollectionTypes       1     True          0
           WMCore_Agent_DefaultConfig       1     True          0
                 WMCore_Configuration       1     True         13
             WMCore_Database_Dialects       1     True          0
          WMCore_GroupUser_Decorators       1     True          0
                        WMCore_ReqMgr       2     True          0
                WMCore_Services_HTTPS       1     True         16
              WMCore_WMSpec_Utilities       1     True          0
 WMCore_WorkQueue_WorkQueueExceptions       1     True          0
         WMCore_FwkJobReport_FileInfo       1     True          2
            WMCore_MicroService_Tools       1     True          5
                WMCore_Services_Rucio       1     True         14
         WMCore_WebTools_FrontEndAuth       1     True          2
               WMCore_ACDC_Collection       1     True          0
                  WMCore_ACDC_Fileset       1     True          0
        WMQuality_Emulators_ReqMgrAux       1     True          1
             WMCore_Agent_ConfigDBMap       1     True          0
    WMCore_Storage_TrivialFileCatalog       1     True          7
      WMCore_WMSpec_ConfigSectionTree       1     True          6
        WMCore_WebTools_DefaultConfig       1     True          0
    WMQuality_Emulators_DefaultConfig       1     True          0
    WMQuality_Emulators_EmulatorSetup       1     True          0
                    WMCore_DAOFactory       1     True          3
          WMCore_Database_Transaction       1     True          0
  WMCore_JobSplitting_SplitterFactory       1     True          0
                 WMCore_WebTools_Page       1     True          3
            WMCore_Database_DBCreator       1     True          9
         WMCore_ResourceControl_MySQL      13     True          6
              WMCore_ThreadPool_MySQL       3     True          0
           WMCore_WMBS_CreateWMBSBase       1     True          2
               WMQuality_TestDB_MySQL       2     True          0
        WMCore_ResourceControl_Oracle      13     True          0
                WMCore_Database_MySQL       2     True          0
               WMCore_Database_Oracle       2     True          0
              WMQuality_TestDB_Oracle       3     True          0
        WMCore_ThreadPool_ThreadSlave       1     True          8
```

#### Is your feature request related to a problem? Please describe.

The end goal is to have WMCore py2 and py3 compatible.

#### Describe the solution you'd like

Run futurize on the first bulk of units and then apply the necessary manual changes.

### PR DESCRIPTION
**Pull request by mapellidario**
Fixes #10098 

#### Status

ready

#### Related PRs

This PR is rebased on `master`.

* Previous migration PR: #10057 (slice1, issue #10055 ). 
* Next migration PR: #10117 (slice3, issue #10114 )

#### Description

Run futurize and some manual changes on the first batch of src/python/A/B/C.

No notable changes or errors have been encountered during this PR.

#### Is it backward compatible (if not, which system it affects?)

It should be. No possible causes for errors have been spot by unit tests. 

#### External dependencies / deployment changes

Requires python-future in both py2 and py3 environments.

### CODE REVIEW

**Comment by amaltaro** [comment]
@mapellidario since this PR has been rebased on top of #10057, I'd rather focus on 10057 and get that one merged first. Then you rebase it once again and many changes/files will be gone, making it a much easier review to be done.
Please correct me if I'm missing anything though.

**Comment by mapellidario** [comment]
I totally agree! Sorry for the noise, thanks again!

**Comment by mapellidario** [comment]
@amaltaro after discussion on slack, this has been rebased directly onto master and it is ready to be review!

File: src/python/WMCore/DAOFactory.py [hunk 1]
```diff
@@ -25,7 +28,7 @@ def __call__(self, classname):
             dia = self.dbinterface.engine.dialect
             #TODO: Make good
             dialect = None
-            for i in self.dialects.keys():
+            for i in list(self.dialects):
```
Comment (amaltaro) [inline src/python/WMCore/DAOFactory.py:31]: It looks like this one could have been:
```
for i in self.dialects:
```

File: src/python/WMCore/Database/DBCreator.py [hunk 2]
```diff
@@ -112,6 +108,6 @@ def __str__(self):
         """
         string = ''
         for i in self.create, self.constraints, self.inserts, self.indexes:
-            for j in i.keys():
+            for j in i:
```
Comment (amaltaro) [inline src/python/WMCore/Database/DBCreator.py:111]: such a horrible variable naming... nothing for you to do though!

**Review by amaltaro — APPROVED** [review]
These changes look good to me, Dario. I assume they are ready to be merged as well?

**Comment by amaltaro** [comment]
Actually, can you please squash commits 2 and 3?

File: src/python/WMCore/DAOFactory.py (hunk 1 above)
Comment (mapellidario) [reply src/python/WMCore/DAOFactory.py:31]: Uh! yes, thanks!

**Comment by mapellidario** [comment]
I squashed commit about `test` and `pylint` and added a commit with your suggestion about `WMCore/DAOFactory.py`. 

After you give the ok, I can squash the commits about `src` and then this is ready to be merged

**Comment by amaltaro** [comment]
Thanks Dario. Feel free to squash the `src` changes and it's ready to go.

**Comment by amaltaro** [comment]
Oups, I overlooked the python2.6 idioms. Have you tried updating those (at the very bottom of the PR report)? Or should we leave it for another PR?

**Comment by mapellidario** [comment]
Since it is the usual `type` vs `isisntance` that broke JSONThunker a couple of months ago, i am a bit scared of touching those, especially when `dict`s are involved. I can give it a try and hope for the best, though

**Comment by amaltaro** [comment]
@mapellidario Dario, I hope you haven't started working on this type x isinstance thing.
As suggested over slack, how about we deal - globally - with this type vs isinstance change in a different PR. I do not expect many modules to be changed and we can nicely isolate that from anything else, making our life easier to spot issues like the JSONThunker one.

**Pull request merged**